pytest_fixtures/core/sys.py

```diff
@@ -47,19 +47,7 @@ def proxy_port_range(default_sat):
 @pytest.fixture
 def install_cockpit_plugin(default_sat):
     default_sat.register_to_dogfood()
-    cmd_result = default_sat.execute(
-        'foreman-installer --enable-foreman-plugin-remote-execution-cockpit', timeout='30m'
-    )
-    if cmd_result.status != 0:
-        raise SatelliteHostError(
-            f'Error during cockpit installation, installation output: {cmd_result.stdout}'
-        )
-    cmd_result = default_sat.execute(
-        f'sshpass -p "{settings.server.ssh_password}" ssh-copy-id \
-        -i ~foreman-proxy/.ssh/id_rsa_foreman_proxy -o StrictHostKeyChecking=no localhost'
-    )
-    if cmd_result.status != 0:
-        raise SatelliteHostError(f'Error during ssh-copy-id, command output: {cmd_result.stdout}')
+    default_sat.install_cockpit()
```

Comment:
`test_positive_cockpit ` should be marked as a destructive test or can be run on separate sat, I don't think it's wise to run the installer on default_sat

Reply:
the question here, is if we are marking the fixtures or we are marking the tests
agree with you in this case all tests with using mentioned fixture should be destructive, so marking fixture seems correct to me

right now, no pytest fixture has destructive marker on it
right now, there is only one test using install cockpit, this test is marked as destructive

Reply:
Interesting, I haven't considered that. Is installing cockpit really a destructive action?

Reply:
no it isn't, is just to make sure, that nothing else run while the installer run, while installer run nothing else should. 
I know we used the marker called run in one thread for that, but not sure how that works now with dynaconf.

Reply:
@tstrych Since the introduction of satellites as xDist workers, we will have one satellite running one test at a time so we dont need that isolation marker at all! 

Though enabling the cockpit plugin on satellite won't affect other tests/components functionality, but IMHO, the tests that alter the default_sat behavior should be treated special cases, in this case, `destructive_sat` similar to katello-agent tests or `destructive` marker to isolate the test is required !

Reply:
@jyejare but altered behavior should not cause any changes, and if it does it is bug, so we should test with altered behavior, understand that such a bug would be hard to find, and it will impact part or whole automation run, but still think it's worth, we are not seeking ever green numbers, we want it red, so we can hunt the problems down and get back to green and again to red colors :)

Reply:
I am not currently under the belief that cockpit installation would be destructive. Altering a Satellite installation in a way that doesn't negatively impact other components shouldn't be considered destructive.

Reply:
@tstrych Yeah So in my last comment I am talking abt that `tests altering the behavior on satellite` should be isolated the way we are doing in `katello-agent` tests or we should find all such tests and make them use the `destructive_sat` fixture so it wont affect anything in any case.

Reply:
and what I am telling you is that your approach, to minimize test result error is great for automation results but it is not for product testing.
so as Jake said while installer actions that doesn't negatively impact satellite should be treated as non destructive

in my  perspective our aim is to get bulletproof product not evergreen automation :) 
maybe I missed your point, and I would be happy to discuss this with you